content/en/Features/db_migration.md

```diff
@@ -0,0 +1,257 @@
+developed to aid in database migrations as well. Often, these features are not just about quality-of-life/usability
+but also about security and recoverability. Database migrations _are dangerous_, and using a home-brewed tool is not
+advised.
+
+For recommended tools, please refer to the documentation of your ORM/client, or even your web framework for cases like
+Flask, Ruby on Rails, etc...
+
+### Ensure Forward/Backwards Compatibility
+One of the least respected, but most important guidelines: always preserve at least one degree of forward/backwards compatibility.
+This rule is not just to help with disaster recovery, but also to maintain the site operational during the migration
+as not all servers may be up-to-date at the time of migration, or if they are that means they were functioning with
+a database one expected migration behind.
+
+For deprecations, the recommended approach is the 3-point migration:
+1. Beginning with your database and server at the original versions, you start by adding any new data/columns but not use them yet.
+2. You introduce a backwards compatible change in the database and/or server where the old data is maintained
+and the new data is used instead.
+3. Once the database and servers are up-to-date, you remove the old, now unused, data from the database.
+
+The 3-point migration is admittedly slower, but it cuts off the vast majority of venues for error.
+
+### Keep an Eye on Database Locks
+
+Depending on the migration tool you are using and the size of the dataset, the migration execution itself could be using 
+long-lasting locks based on the amount of data being created/updated/destroyed. This can create serious problems as the
+locks may forbid any parallel write operations, meaning your application will fail in any endpoint executing such an
+action, and cause downtime. These dangers are more difficult to track as they are typically dependent on the size of the
+database, and as staging databases are typically a fraction of the size of production, such QA will not catch the danger.
+Aside from a full production replication for testing, the only reasonable deterrent would be a database-educated engineer
+approving/guarding db migration addition.
+
+### Validate on as Valid as Data as Possible Outside of Production
+
+Even in small datasets, database migrations will always face the danger of improper data transformations which cause errors.
+One of the best countermeasures for such errors is testing the migrations in non-production databases which closely
+resemble production-- the more similar the better as it will test more potential edge cases. Typically, this is done
+as part of the deployment to a "staging/qa" environment.
+
+### Have Folks on Standby
+
+Always prepare under the assumption that if something can go wrong, it will go wrong. Hence, always have someone on 
+standby monitoring the migration.
```

Comment:
this is a good point, but this is also very common to run DB migrations when there is low traffic - so typically at night. 
So the end goal is to have a rollout mechanism and more testing vs having people in standby for each migration. May be for unusual large migrations, have folks in standby makes more sense.